pkg/project/admission/lifecycle/admission.go

```diff
@@ -140,10 +141,26 @@ func (e *lifecycle) Handles(operation admission.Operation) bool {
 }
 
 func NewLifecycle(client client.Interface, creatableResources sets.String) (admission.Interface, error) {
```

Comment:
Pass the cache through here and then call this directly.  We already have to do this for the `ServiceAccount` admission plugin and this global variable hid this problem from us.

@derekwaynecarr @csrwng Is there a cleaner way to pass a specific instance of something through?  There's been talk of changing the plugin structure.  I think this is a good example of a common problem that we have.